pyhive/tests/test_hive.py

```diff
@@ -205,13 +202,20 @@ def test_custom_transport(self):
         sasl_auth = 'PLAIN'
 
         def sasl_factory():
-            sasl_client = sasl.Client()
-            sasl_client.setAttr('host', 'localhost')
-            sasl_client.setAttr('username', 'test_username')
-            sasl_client.setAttr('password', 'x')
-            sasl_client.init()
-            return sasl_client
-
+            try:
+                import sasl
```

Comment:
let's check if the module is installed then use it's implementation rather than catch import error